pkg/cli/scan.go

```diff
@@ -205,19 +208,36 @@ wolfictl scan package1 package2 --remote
 
 			// Do a scan for each input
 
-			var scans []inputScanResult
+			var g errgroup.Group
+			g.SetLimit(runtime.GOMAXPROCS(0))
+
+			scans := make([]inputScanResult, len(inputs))
 			var inputPathsFailingRequireZero []string
-			for _, input := range inputs {
-				isr, err := p.doScanCommandForSingleInput(cmd.Context(), input, advisoryDocumentIndices)
-				if err != nil {
-					return err
-				}
-				scans = append(scans, *isr)
+			for i, input := range inputs {
+				i, input := i, input
 
-				if p.requireZeroFindings && len(isr.Result.Findings) > 0 {
-					// Accumulate the list of failures to be returned at the end, but we still want to complete all scans
-					inputPathsFailingRequireZero = append(inputPathsFailingRequireZero, input)
-				}
+				g.Go(func() error {
+					isr, err := p.doScanCommandForSingleInput(cmd.Context(), input, advisoryDocumentIndices)
+					if err != nil {
+						return err
+					}
+
+					scans[i] = *isr
```

Comment:
Are we sure this kind of slice access is thread-safe?

Reply:
Yep! Appending isn't, which is why below on findings I use the lock to append to `inputPathsFailingRequireZero`.